docs/beginner/dict_vs_defaultdict.md

```diff
@@ -0,0 +1,56 @@
+# dict vs defaultdict
+
+Inspect the difference between a `dict` and a `defaultdict` in Python.
+
+## Access dict by key with square brackets
+
+The easiest way to access a value inside dictionary is with **[key]** syntax.
+But be careful, if the key does not exist, you will get a `KeyError`.
+
+::: src.beginner.dict_vs_defaultdict
+    options:
+      members:
+        - get_value_from_key_with_square_brackets
+
+## Access dict by key with get method
+
+Using `get` method is another option. If key is not present in dict, None (or
+custom value) is returned.
+
+::: src.beginner.dict_vs_defaultdict
+    options:
+      members:
+        - get_value_from_key_with_get
+
+## Use defaultdict
+
+**defaultdict** enables a dict with a default value. Useful in the occasions
+where you need a default value always. String, int, list, None... any type you want.
```

Comment:
defaultdict has a parameter, default_factory,  which is not an string. It is a function that return a value in case of the given value is not in the dict.
if the default_factory is not provided, the dictionary will return a KeyError exception.